flax/linen/fp8_ops.py

```diff
@@ -306,18 +317,27 @@ def __call__(self, *args, **kwargs):
     comp_dtype = k.dtype
     x = jnp.asarray(x, comp_dtype)
 
+    e4m3_dtype, e5m2_dtype = get_fp8_dtypes(self.fp8_genre)
+
     x_qdq = in_qdq(
-      comp_dtype, x, self.input_scale.value, self.input_amax_history.value
+      comp_dtype, e4m3_dtype, x, self.input_scale.value, self.input_amax_history.value
     )
     k_qdq = in_qdq(
-      comp_dtype, k, self.kernel_scale.value, self.kernel_amax_history.value
+      comp_dtype, e4m3_dtype, k, self.kernel_scale.value, self.kernel_amax_history.value
     )
     y_qdq = dot_general_with_precision(x_qdq, k_qdq, dimension_numbers)  # type: ignore
     y = out_qdq(
       comp_dtype,
+      e5m2_dtype,
       y_qdq,
       self.output_grad_scale.value,
       self.output_grad_amax_history.value,
     )
 
     return y  # type: ignore
+
+class OCPFp8DotGeneralOp(Fp8DotGeneralOp):
+  fp8_genre: str = 'OCP'
+
+class NANOOFp8DotGeneralOp(Fp8DotGeneralOp):
+  fp8_genre: str = 'NANOO'
```

Comment:
e.g. this would now be:
```
class NANOOFp8DotGeneralOp(Fp8DotGeneralOp):
  e4m3_dtype: Dtype = jnp.float8_e4m3fnuz
  e5m2_dtype: Dtype = jnp.float8_e5m2nuz
```

Reply:
Done.